openBCIBoard.js

```diff
@@ -856,180 +1013,174 @@ function OpenBCIFactory() {
+            Sntp.offset(function(err, offset) {
+                if(err) reject(err);
+                resolve(offset);
+            });
+        });
     };
 
     /**
-     * @description Strip packets from the master buffer
-     * @returns {Buffer} A buffer containing a packet of 33 bytes long, ready to be read.
+     * @description Get time from the NTP server. Must have internet connection!
+     * @returns {Promise} - Fulfilled with time object
      * @author AJ Keller (@pushtheworldllc)
      */
-    OpenBCIBoard.prototype._bufPacketStripper = function() {
-        try {
-            // not at end of master buffer
-            var rawPacket;
-            if(k.OBCIPacketSize < k.OBCIMasterBufferSize - this.masterBuffer.positionRead) {
-                // extract packet
-                rawPacket = this.masterBuffer.buffer.slice(this.masterBuffer.positionRead, this.masterBuffer.positionRead + k.OBCIPacketSize);
-                // move the read position pointer
-                this.masterBuffer.positionRead += k.OBCIPacketSize;
-                // increment packets read
-                this.masterBuffer.packetsRead++;
-                //console.log(rawPacket);
-                // return this raw packet
-                return rawPacket;
-            } else { //special case because we are at the end of the master buffer (must wrap)
-                // calculate the space left to read from for the partial packet
-                var part1Size = k.OBCIMasterBufferSize - this.masterBuffer.positionRead;
-                // make the first part of the packet
-                var part1 = this.masterBuffer.buffer.slice(this.masterBuffer.positionRead, this.masterBuffer.positionRead + part1Size);
-                // reset the read position to 0
-                this.masterBuffer.positionRead = 0;
-                // get part 2 size
-                var part2Size = k.OBCIPacketSize - part1Size;
-                // get the second part
-                var part2 = this.masterBuffer.buffer.slice(0, part2Size);
-                // merge the two parts
-                rawPacket = Buffer.concat([part1,part2], k.OBCIPacketSize);
-                // move the read position pointer
-                this.masterBuffer.positionRead += part2Size;
-                // increment packets read
-                this.masterBuffer.packetsRead++;
-                // return this raw packet
-                return rawPacket;
-            }
-        }
-        catch (error) {
-            console.log('Error: ' + error);
-        }
+    OpenBCIBoard.prototype.sntpGetServerTime = function() {
+        return new Promise((resolve,reject) => {
+            Sntp.time(this.ntpOptions, function (err, time) {
+
+                if (err) {
+                    console.log('Failed: ' + err.message);
+                    reject(err);
+                    //process.exit(1);
+                }
+
+                console.log('Local clock is off by: ' + time.t + ' milliseconds');
```

Comment:
where is `time.t` being defined? it must be getting the board's from somewhere. I don't see the call for server time.